WordPress/Classes/ViewRelated/Blog/Site Picker/BlogList/BlogListViewModel.swift

```diff
@@ -20,184 +17,104 @@ final class BlogListViewModel: NSObject, ObservableObject {
     ) {
         self.contextManager = contextManager
         self.eventTracker = eventTracker
+        self.fetchedResultsController = createFetchedResultsController(in: contextManager.mainContext)
         super.init()
-        setupFetchedResultsControllers()
+        setupFetchedResultsController()
     }
 
     func searchQueryChanged(_ newText: String) {
         searchText = newText
         updateSearchResults()
     }
 
-    func togglePinnedSite(siteID: NSNumber?) {
-        guard let siteID, let blog = allBlogs.first(where: { $0.dotComID == siteID }) else {
-            return
-        }
-        let isCurrentlyPinned = blog.pinnedDate != nil
-
-        trackPinned(blog: blog)
-
-        blog.pinnedDate = isCurrentlyPinned ? nil : Date()
-
-        contextManager.saveContextAndWait(contextManager.mainContext)
-    }
-
-    func siteSelected(siteID: NSNumber?) {
-        guard let siteID, let blog = allBlogs.first(where: { $0.dotComID == siteID }) else {
-            return
+    func didSelectSite(withSiteID siteID: NSNumber) -> Blog? {
+        guard let blog = rawSites.first(where: { $0.dotComID == siteID }) else {
+            return nil
         }
-
-        trackSiteSelected(blog: blog)
-
+        eventTracker.track(.siteSwitcherSiteTapped, properties: [
+            "section": blog.lastUsed != nil ? "recent" : "all"
+        ])
         blog.lastUsed = Date()
-
         contextManager.saveContextAndWait(contextManager.mainContext)
-    }
-
-    private func trackPinned(blog: Blog) {
-        eventTracker.track(
-            .siteSwitcherPinUpdated,
-            properties: [
-                "blog_id": blog.dotComID ?? "",
-                "pinned": blog.pinnedDate == nil
-            ]
-        )
-    }
-
-    private func trackSiteSelected(blog: Blog) {
-        let sectionName: String
-
-        if blog.pinnedDate != nil {
-            sectionName = "pinned"
-        } else if blog.lastUsed != nil {
-            sectionName = "recent"
-        } else {
-            sectionName = "all"
-        }
-
-        eventTracker.track(
-            .siteSwitcherSiteTapped,
-            properties: [
-                "section": sectionName,
-            ]
-        )
+        return blog
     }
 
     func viewAppeared() {
```

Comment:
If the view disappears, should we cancel all in-flight tasks?

Reply:
There are no long-running tasks in the ViewModel in this PR. It doesn't sync the sites yet.
I added syncing in the following PR https://github.com/wordpress-mobile/WordPress-iOS/pull/23432/files#diff-472de19e113614cf6d4ee31909aead1fb2e65eaee973e3be078f482f35cdfe68R44 and it does cancel it on disappear.